lib/index.js

```diff
@@ -326,6 +325,36 @@
     // Changed to stroke to match other stroke functions
     this.setStrokeDash = this.setLineDash;
 
+    this.fill = function () {
+      var args = Array.prototype.slice.call(arguments);
+
+      if (args.length === 1 && typeof args[0] === 'string' && !isSpecialFill.exec(args[0])) {
+        this.setFill(args[0]);
+        ctx.fill();
+        return this;
+      } else if (args.length === 2 && typeof args[1] === 'string' && !isSpecialFill.exec(args[1])) {
+        this.setFill(args[1]);
+        ctx.fill(args[0]);
+        return this;
+      }
```

Comment:
PErsonally i'd extract helper functions to reduce duplication and make meaning clearer but YMMV:

```javascript
function fillDescriptionIsArg(n) {
  return args.length === n+1 && typeof args[n] === 'string' && !isSpecialFill.exec(args[n]);
}

function fillWith(fillDescription, fill) {
  this.setFill(fillDescription);
  ctx.fill(fill);
  return this;
}

if (fillDescriptionIsArg(0)) {
  return fillWith(args[0]);
} else if (fillDescriptionIsArg(1)) {
  return fillWith(args[1], args[0]);
}

```